include/alpaka/dev/DevOacc.hpp

```diff
@@ -0,0 +1,343 @@
+        -> std::vector<std::shared_ptr<IGenericThreadsQueue<DevOacc>>>
+        {
+            return m_spDevOaccImpl->getAllExistingQueues();
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Registers the given queue on this device.
+        //! NOTE: Every queue has to be registered for correct functionality of device wait operations!
+        ALPAKA_FN_HOST auto registerQueue(std::shared_ptr<IGenericThreadsQueue<DevOacc>> spQueue) const
+        -> void
+        {
+            m_spDevOaccImpl->registerQueue(spQueue);
+        }
+
+    public:
+        std::shared_ptr<oacc::detail::DevOaccImpl> m_spDevOaccImpl;
+    };
+
+    namespace traits
+    {
+        //#############################################################################
+        //! The OpenACC device name get trait specialization.
+        template<>
+        struct GetName<
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getName(
+                DevOacc const &)
+            -> std::string
+            {
+                return std::string("OpenACC target");
+            }
+        };
+
+        //#############################################################################
+        //! The OpenACC device available memory get trait specialization.
+        template<>
+        struct GetMemBytes<
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getMemBytes(
+                DevOacc const & dev)
+            -> std::size_t
+            {
+                return acc_get_property(dev.iDevice(), dev.deviceType(), acc_property_memory);
+            }
+        };
+
+        //#############################################################################
+        //! The OpenACC device free memory get trait specialization.
+        template<>
+        struct GetFreeMemBytes<
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getFreeMemBytes(
+                DevOacc const & dev)
+            -> std::size_t
+            {
+                return acc_get_property(dev.iDevice(), dev.deviceType(), acc_property_free_memory);
+            }
+        };
+
+        //#############################################################################
+        //! The OpenACC device warp size get trait specialization.
+        template<>
+        struct GetWarpSize<
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getWarpSize(
+                DevOacc const & dev)
+            -> std::size_t
+            {
+                alpaka::ignore_unused(dev);
```

Comment:
If `dev` is not used, then by declare it in the first place? You can just have `DevOacc const&)` in the parameter list.

Reply:
But then you would lose the parameter name completely which may not be desired because it might contain usage hints or because we may want to add doxygen documentation for the parameter. Both `alpaka::ignore_unused` as well as removing the name are not the right solution. The only correct solution is the c++17 `maybe_unused` attribute which we can not use yet. For now `alpaka::ignore_unused` is the consisten way.

Reply:
I know these reasons, but there is neither a doxygen comment nor does the name `dev` help here ;) But we can leave it as is.